src/hyperlight_host/src/sandbox/initialized_multi_use.rs

```diff
@@ -59,17 +59,17 @@ static SANDBOX_ID_COUNTER: AtomicU64 = AtomicU64::new(0);
 ///
 /// Only one guard can exist per interrupt handle at a time - attempting to create
 /// a second guard will return an error.
-struct CallActiveGuard {
-    interrupt_handle: Arc<dyn InterruptHandle>,
+struct CallActiveGuard<T: crate::hypervisor::InterruptHandleInternal + ?Sized> {
+    interrupt_handle: Arc<T>,
 }
 
-impl CallActiveGuard {
+impl<T: crate::hypervisor::InterruptHandleInternal + ?Sized> CallActiveGuard<T> {
```

Comment:
Is this change for performance reasons?

Reply:
No, I think it can be simplified to take `dyn InterruptHandleInternal` rather than be generic